tests/test_mirrors.py

```diff
@@ -60,30 +60,36 @@ def setUp(self):
                  'metadata_path' : 'metadata',
                  'targets_path' : 'targets',
                  'confined_target_dirs' : ['targets/release/',
-                                            'targets/release/']}}
+                                            'targets/release/']},
+     # Make sure we are testing if confined_target_dirs is ['']
```

Comment:
```suggestion
     # Make sure we are testing when confined_target_dirs is ['']
```

I wonder if this is overloading the current test too much, perhaps it would be cleaner to add a new test case just for the `confined_target_dirs = ['']` scenario?

Reply:
This test doesn't contain complicated logic it just contains multiple test cases.
If we are going to add a separate test for this case with the new mirror then it makes sense to split the whole current test case into 4 or 5 tests.